ocsgw/src/main/java/org/ostelco/ocsgw/datasource/protobuf/PubSubDataSource.kt

```diff
@@ -119,8 +119,10 @@ class PubSubDataSource(
             override fun onFailure(throwable: Throwable) {
                 if (throwable is ApiException) {
                     // details on the API exception
-                    logger.warn("Status code: {}", throwable.statusCode.code)
-                    logger.warn("Retrying: {}", throwable.isRetryable)
+                    logger.warn("Pubsub topic: $ccaTopicId\n" +
+                            "Message : ${throwable.message}\n" +
+                            "Status code: ${throwable.statusCode.code}\n" +
+                            "Retrying: ${throwable.isRetryable}")
                 }
                 logger.warn("Error sending CCR Request to PubSub")
```

Comment:
Put this in `else` with Topic ID to avoid duplicate errors.

Reply:
done